executor/executor_x.go

```diff
@@ -0,0 +1,341 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"math"
+	"sort"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/optimizer/plan"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/types"
+	"github.com/pingcap/tidb/xapi"
+	"github.com/pingcap/tidb/xapi/tablecodec"
+	"github.com/pingcap/tidb/xapi/tipb"
+)
+
+// XSelectTableExec represents XAPI select executor.
+type XSelectTableExec struct {
+	table     table.Table
+	tablePlan *plan.TableScan
+	where     *tipb.Expr
+	ctx       context.Context
+	result    *xapi.SelectResult
+	subResult *xapi.SubResult
+}
+
+// Next implements Executor Next interface.
+func (e *XSelectTableExec) Next() (*Row, error) {
+	if e.result == nil {
+		err := e.doRequest()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+	for {
+		if e.subResult == nil {
+			var err error
+			e.subResult, err = e.result.Next()
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			if e.subResult == nil {
+				return nil, nil
+			}
+		}
+		h, rowData, err := e.subResult.Next()
```

Comment:
As I know, MySQL Cluster has pushed partial `Join` operations down, and gained some benefits.